tests/Fixtures/Integration/misc/PHP8_0.test

```diff
@@ -0,0 +1,367 @@
+    {
+    }
+}
+
+// https://wiki.php.net/rfc/throw_expression
+if (1) {
+    foo() ?? throw new \Exception();
+    $a = $condition && throw new Exception();
+    $callable = fn () => throw new Exception();
+    $value = $falsableValue ?: throw new InvalidArgumentException();
+}
+
+// https://wiki.php.net/rfc/non-capturing_catches
+// TODO: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5175 - when implementing new rule for fixing such cases, add exception variable in --INPUT-- section
+try {
+    foo();
+} catch (Exception) {
+    // ignore exception
+}
+
+// https://wiki.php.net/rfc/constructor_promotion
+class Point
+{
+    public function __construct(
+        public float $x = 0.0,
+        protected float $y = 0.0,
+        private float $z = 0.0,
+    ) {
+    }
+}
+
+// https://wiki.php.net/rfc/match_expression_v2
+echo match ($x) {
+    1, 2 => 'Same for 1 and 2',
+};
+
+// https://wiki.php.net/rfc/add_str_starts_with_and_ends_with_functions
+// no syntax changes
+
+// https://wiki.php.net/rfc/str_contains
+// no syntax changes
+
+// https://wiki.php.net/rfc/deprecate_curly_braces_array_access
+// syntax was deprecated, nothing we can do - a `normalize_index_brace` rule already exists and can be used on PHP <8 (or PHP 8 if errors are ignored)
+
+// https://wiki.php.net/rfc/concatenation_precedence
+// nothing we can do
+
+// https://wiki.php.net/rfc/inheritance_private_methods
+// nothing we can do
+
+// https://wiki.php.net/rfc/ternary_associativity
+// nothing we can do about the warning - a rule can be implemented to run on PHP <8 (or PHP 8 if errors are ignored)
+
+// https://wiki.php.net/rfc/stringable
+// nothing we can do
+
+// Date: mktime() and gmmktime() now require at least one argument. time() can be used to get the current timestamp.
+time();
+time();
+mktime($a);
+gmmktime(1, 2, 3, 4, 5, 6);
+
+// Exif: Removed read_exif_data(). exif_read_data() should be used instead.
+exif_read_data($filename, $sections_needed, $sub_arrays, $read_thumbnail);
+
+// The imap_header() function which is an alias of imap_headerinfo() has been removed.
+imap_headerinfo();
+
+// parse_str() can no longer be used without specifying a result array.
+// nothing we can do
+
+// Calling implode() with parameters in a reverse order ($pieces, $glue) is no longer supported.
+implode('', $pieces);
+implode('', $pieces);
+
+// Mbstring A number of deprecated mbregex aliases have been removed.
+// nothing we can do
+
+// Namespaced names can no longer contain whitespace (nor comments)
+// nothing we can do
+
+--INPUT--
+<?php
+
+// https://wiki.php.net/rfc/named_params
+array_fill(start_index: 0, num: 100, value: 50);
+htmlspecialchars($string, double_encode: false);
+array_slice($array, $offset, $length, preserve_keys: true);
+new ParamNode($name, variadic: $isVariadic, byRef: $passByRef);
+array_foobar(array: $value); // reserved keyword as parameter name!
+
+// https://wiki.php.net/rfc/nullsafe_operator
+// TODO - CRASHING !!!
```

Comment:
@SpacePossum , can you help with this crashing code?

Reply:
I get the linter error here because of https://bugs.php.net/bug.php?id=80462 , there is not much we can do until the fix gets released by PHP team I think

Reply:
imho, that means we should forbid 8.0 and allow only 8.0.x, whenever the fix is released

Reply:
:+1:  agreed, I hope they release soon

Reply:
PHP 8.0.1 is scheduled to be released on January 7th.

Reply:
ref https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5415